src/ai-bundle/src/AiBundle.php

```diff
@@ -440,42 +462,46 @@ private function processAgentConfig(string $name, array $config, ContainerBuilde
 
                 $toolProcessorDefinition = (new ChildDefinition('ai.tool.agent_processor.abstract'))
                     ->replaceArgument(0, new Reference('ai.toolbox.'.$name));
-                $container->setDefinition('ai.tool.agent_processor.'.$name, $toolProcessorDefinition);
 
-                $inputProcessors[] = new Reference('ai.tool.agent_processor.'.$name);
-                $outputProcessors[] = new Reference('ai.tool.agent_processor.'.$name);
+                $container->setDefinition('ai.tool.agent_processor.'.$name, $toolProcessorDefinition)
+                    ->addTag('ai.agent.input_processor', ['agent' => $name, 'priority' => -10])
+                    ->addTag('ai.agent.output_processor', ['agent' => $name, 'priority' => -10]);
             } else {
                 if ($config['fault_tolerant_toolbox'] && !$container->hasDefinition('ai.fault_tolerant_toolbox')) {
                     $container->setDefinition('ai.fault_tolerant_toolbox', new Definition(FaultTolerantToolbox::class))
                         ->setArguments([new Reference('.inner')])
                         ->setDecoratedService('ai.toolbox');
                 }
 
-                $inputProcessors[] = new Reference('ai.tool.agent_processor');
-                $outputProcessors[] = new Reference('ai.tool.agent_processor');
+                $container->getDefinition('ai.tool.agent_processor')
+                    ->addTag('ai.agent.input_processor', ['agent' => $name, 'priority' => -10])
```

Comment:
The main discussion might be about the priority of the registered processor @chr-hertel.

By default custom processor are registered with the 0 priority meaning they are applied BEFORE all the processors from Symfony.
If we want the opposite, I need to use a positive priority here.